lib/ConstExtract/ConstExtract.cpp

```diff
@@ -119,28 +120,68 @@ parseProtocolListFromFile(StringRef protocolListFilePath,
   return true;
 }
 
+static std::string extractLiteralOutput(Expr *expr) {
+  std::string LiteralOutput;
+  llvm::raw_string_ostream OutputStream(LiteralOutput);
+  expr->printConstExprValue(&OutputStream, nullptr);
+
+  return LiteralOutput;
+}
+
 static std::shared_ptr<CompileTimeValue>
 extractPropertyInitializationValue(VarDecl *propertyDecl) {
   auto binding = propertyDecl->getParentPatternBinding();
   if (binding) {
     auto originalInit = binding->getOriginalInit(0);
     if (originalInit) {
-      std::string LiteralOutput;
-      llvm::raw_string_ostream OutputStream(LiteralOutput);
-      originalInit->printConstExprValue(&OutputStream, nullptr);
-      if (!LiteralOutput.empty())
+      auto LiteralOutput = extractLiteralOutput(originalInit);
+      if (!LiteralOutput.empty()) {
         return std::make_shared<RawLiteralValue>(LiteralOutput);
+      }
+
+      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
+        std::vector<FunctionParameter> parameters;
+
+        const auto args = callExpr->getArgs();
+        for (auto arg : *args) {
+          auto label = arg.getLabel().str().str();
```

Comment:
Why the double `.str()` here?

Reply:
The first `.str()` returns a `StringRef` so we need an additional `.str()` to get the `std::string`.

```cpp
swift::Identifier identifier = arg.getLabel();
StringRef stringRef = identifier.str();
std::string label = stringRef.str();
```

Reply:
Interesting! Makes sense now.